packages/mcp-client/src/McpClient.ts

```diff
@@ -47,20 +53,42 @@ export class McpClient {
 		model: string;
 		apiKey: string;
 	}) {
-		this.client = endpointUrl ? new InferenceClient(apiKey, { endpointUrl: endpointUrl }) : new InferenceClient(apiKey);
+		const clientOptions = endpointUrl ? ({ endpointUrl } as Options & { endpointUrl: string }) : undefined;
+		this.client = endpointUrl ? new InferenceClient(apiKey, clientOptions) : new InferenceClient(apiKey);
 		this.provider = provider;
 		this.model = model;
 	}
-
-	async addMcpServers(servers: StdioServerParameters[]): Promise<void> {
+	async addMcpServers(servers: (ServerConfig | StdioServerParameters)[]): Promise<void> {
 		await Promise.all(servers.map((s) => this.addMcpServer(s)));
 	}
 
-	async addMcpServer(server: StdioServerParameters): Promise<void> {
-		const transport = new StdioClientTransport({
-			...server,
-			env: { ...server.env, PATH: process.env.PATH ?? "" },
-		});
+	async addMcpServer(server: ServerConfig | StdioServerParameters): Promise<void> {
+		let transport: Transport;
+		const asUrl = (url: string | URL): URL => {
+			return typeof url === "string" ? new URL(url) : url;
+		};
+
+		if (!("type" in server)) {
+			transport = new StdioClientTransport({
+				...server,
+				env: { ...server.env, PATH: process.env.PATH ?? "" },
```

Comment:
ah noo it's here that i needed to pass the env.PATH explicitly! maybe it's fixed?